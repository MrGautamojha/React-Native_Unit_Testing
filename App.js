import * as React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import StackNavigator from './src/navigation/stackNavigation';
import { navigationRef, routeNameRef } from './src/services/navigationService';
import { Provider, useSelector } from 'react-redux';
import store from './src/redux/store'
import { View } from 'react-native';


function App() {


  return (
    <View testID={'APPTEST'}>
      <Provider store={store} >
        <NavigationContainer
          ref={navigationRef}
        >
          <StackNavigator />
        </NavigationContainer>
      </Provider>
    </View>
  );
}

export default App;
