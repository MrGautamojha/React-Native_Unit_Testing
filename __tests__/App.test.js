/**
 * @format
 */
 import React from 'react';
 import { shallow } from "enzyme";
 import  App  from "../App";
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('App', function () {
  it('App', function () {
    const wrapper = shallow(<App/>);
    expect(wrapper).toMatchSnapshot();
  });
});
