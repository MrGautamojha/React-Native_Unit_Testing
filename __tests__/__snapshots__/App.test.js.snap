// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App 1`] = `
<View
  testID="APPTEST"
>
  <Provider
    store={
      Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <ForwardRef(NavigationContainerInner)>
      <StackNavigator />
    </ForwardRef(NavigationContainerInner)>
  </Provider>
</View>
`;
